WebApiCoreSeed.WebApi.IntegrationTests/Users/UserIntegrationTest.cs

```diff
@@ -0,0 +1,285 @@
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using WebApiCoreSeed.Data.EF;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.WebApi.IntegrationTests.Fake;
+using WebApiCoreSeed.WebApi.IntegrationTests.Users.TestData;
+using WebApiCoreSeed.WebApi.Models;
+using Xunit;
+
+namespace WebApiCoreSeed.WebApi.IntegrationTests.Users
+{
+    public class UserIntegrationTest
+    {
+        private const string ResouceUri = "api/users/";
+
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        private readonly DbContextOptions<WebApiCoreSeedContext> _dbContextOptions;
+
+        public UserIntegrationTest()
+        {
+            FileFaker.Fake();
+
+            var builder = new WebHostBuilder()
+                .UseEnvironment("development")
+                .UseStartup<Startup>();
+
+            _server = new TestServer(builder);
+            _client = _server.CreateClient();
+            _dbContextOptions = _server.Host.Services
+                .GetService(typeof(DbContextOptions<WebApiCoreSeedContext>)) as DbContextOptions<WebApiCoreSeedContext>;
+        }
+
+        #region Get Tests
+
+        [Fact]
+        public async Task Get_ShouldReturnAListOfUsers_WhenNoParameters()
+        {
+            var result = await _client.GetAsync(ResouceUri);
+            result.EnsureSuccessStatusCode();
+
+            var users = JsonConvert.DeserializeObject<List<User>>(await result.Content.ReadAsStringAsync());
+            Assert.NotNull(users);
+
+            using (var dbcontext = CreateContext())
+            {
+                Assert.True(dbcontext.Users.Any(a => users.Any(s => a.Id == s.Id)));
+            }
```

Comment:
This is good but testing against the database will take more time to develop and to execute. For endpoints like a GET you can safely ensure it returns the HTTP status code and the data you expect. Let's keep db verifications when we don't receive an output.